From 9261933081e5af4f693cda5c9b8c32a51bab78e9 Mon Sep 17 00:00:00 2001
From: Robert Bogart <robert.bogart@joyent.com>
Date: Wed, 11 Apr 2018 22:45:40 +0000
Subject: [PATCH] MANTA-3186 Muskie should always log remoteAddress Reviewed
 by: Jordan Hendricks <jordan.hendricks@joyent.com> Reviewed by: Jan Wyszynski
 <jan.wyszynski@joyent.com> Approved by: Jan Wyszynski
 <jan.wyszynski@joyent.com>

---
 lib/audit.js  | 2 +-
 lib/common.js | 9 +++++++++
 2 files changed, 10 insertions(+), 1 deletion(-)

diff --git a/lib/audit.js b/lib/audit.js
index b6f3e52..ccd0829 100644
--- a/lib/audit.js
+++ b/lib/audit.js
@@ -227,7 +227,7 @@ function auditLogger(options) {
              * Otherwise, it will be the same as remoteAddress.
              */
             logicalRemoteAddress: req.connection._xff,
-            remoteAddress: req.connection.remoteAddress,
+            remoteAddress: req.remoteAddress,
             remotePort: req.connection.remotePort,
             req_id: req.id,
             reqHeaderLength: reqHeaderLength,
diff --git a/lib/common.js b/lib/common.js
index 864f9dc..c4d3aed 100644
--- a/lib/common.js
+++ b/lib/common.js
@@ -832,6 +832,15 @@ module.exports = {
                 }
             }
 
+            /*
+             * This might seem over-gratuitous, but it's necessary.  Per the
+             * node.js documentation, if the socket is destroyed, it is possible
+             * for `remoteAddress' to be undefined later on when we attempt to
+             * log the specifics around this request.  As an insurance policy
+             * against that, save off the remoteAddress now.
+             */
+            req.remoteAddress = req.connection.remoteAddress;
+
             var ua = req.headers['user-agent'];
             if (ua && /^curl.+/.test(ua))
                 res.set('Connection', 'close');
-- 
2.21.0

